src/Og.php

```diff
@@ -169,6 +171,31 @@ public static function getEntityGroups($entity_type, $entity_id, $states = [OG_S
   }
 
   /**
+   * Return TRUE if entity belongs to a group.
+   *
+   * @param \Drupal\Core\Entity\EntityInterface $group
+   *   The group entity to check.
+   * @param \Drupal\Core\Entity\EntityInterface $entity
+   *   The entity to get groups for.
+   * @param array $states
+   *   (optional) Array with the membership states to check the membership.
+   *   Defaults to active memberships.
+   *
+   * @return bool
+   *   TRUE if the entity (e.g. the user) belongs to a group and is not pending
+   *   or blocked.
+   */
+  public static function isMember(EntityInterface $group, EntityInterface $entity, $states = [OG_STATE_ACTIVE]) {
```

Comment:
Exactly! We need a proper discussion about our services I think. Now we
have a lot of the building blocks there I think soon could be a good time.
On 7 Dec 2015 18:30, "Daniel Wehner" <notifications@github.com> wrote:

> In src/Og.php
> <https://github.com/amitaibu/og/pull/53#discussion_r46856897>:
>
> > @@ -169,6 +171,31 @@ public static function getEntityGroups($entity_type, $entity_id, $states = [OG_S
> >    }
> >
> >    /**
> > +   * Return TRUE if entity belongs to a group.
> > +   *
> > +   * @param \Drupal\Core\Entity\EntityInterface $group
> > +   *   The group entity to check.
> > +   * @param \Drupal\Core\Entity\EntityInterface $entity
> > +   *   The entity to get groups for.
> > +   * @param array $states
> > +   *   (optional) Array with the membership states to check the membership.
> > +   *   Defaults to active memberships.
> > +   *
> > +   * @return bool
> > +   *   TRUE if the entity (e.g. the user) belongs to a group and is not pending
> > +   *   or blocked.
> > +   */
> > +  public static function isMember(EntityInterface $group, EntityInterface $entity, $states = [OG_STATE_ACTIVE]) {
>
> Fair point regarding static methods, but yeah having kernel tests helps a
> lot of with refactoring those things
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/amitaibu/og/pull/53/files#r46856897>.
>